ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,270 @@
+        }
+
+        if #available(iOS 13.4, *) {
+            // Workaround check for beta iOS versions missing the Pointer Interactions API
+            if arePointerInteractionAPIsAvailable() {
+                button.isPointerInteractionEnabled = true
+            }
+        }
+    }
+
+    private func updateBadgeView() {
+        badgeView.text = badgeValue
+        badgeView.isHidden = badgeValue == nil
+
+        if badgeValue != nil {
+            let maskLayer = CAShapeLayer()
+            maskLayer.fillRule = .evenOdd
+
+            let path = UIBezierPath(rect: CGRect(x: 0, y: 0, width: button.frame.size.width, height: button.frame.size.height))
+            let badgeVerticalOffset = isInPortraitMode ? Constants.badgePortraitVerticalOffset : Constants.badgeLandscapeVerticalOffset
+
+            if badgeView.text?.count ?? 1 > 1 {
+                let badgeWidth = min(max(badgeView.intrinsicContentSize.width + Constants.badgeHorizontalPadding, Constants.badgeMinWidth), Constants.badgeMaxWidth)
+
+                badgeView.frame = CGRect(x: badgeFrameOriginX(frameWidth: badgeWidth),
+                                         y: button.frame.origin.y + badgeVerticalOffset,
+                                         width: badgeWidth,
+                                         height: Constants.badgeHeight)
+
+                let layer = CAShapeLayer()
+                layer.path = UIBezierPath(roundedRect: badgeView.bounds,
+                                          byRoundingCorners: .allCorners,
+                                          cornerRadii: CGSize(width: Constants.badgeCorderRadii, height: Constants.badgeCorderRadii)).cgPath
+
+                path.append(UIBezierPath(roundedRect: badgeBorderRect(badgeViewFrame: badgeView.frame),
+                                         byRoundingCorners: .allCorners,
+                                         cornerRadii: CGSize(width: Constants.badgeCorderRadii, height: Constants.badgeCorderRadii)))
+
+                badgeView.layer.mask = layer
+                badgeView.layer.cornerRadius = 0
+            } else {
+                let badgeWidth = max(badgeView.intrinsicContentSize.width, Constants.badgeMinWidth)
+
+                badgeView.frame = CGRect(x: badgeFrameOriginX(frameWidth: badgeWidth),
+                                         y: button.frame.origin.y + badgeVerticalOffset,
+                                         width: badgeWidth,
+                                         height: Constants.badgeHeight)
+
+                path.append(UIBezierPath(ovalIn: badgeBorderRect(badgeViewFrame: badgeView.frame)))
+
+                badgeView.layer.mask = nil
+                badgeView.layer.cornerRadius = badgeWidth / 2
+            }
+
+            maskLayer.path = path.cgPath
+            button.layer.mask = maskLayer
+        } else {
+            button.layer.mask = nil
+        }
+    }
+
+    private func badgeFrameOriginX(frameWidth: CGFloat) -> CGFloat {
```

Comment:
in your demo screenshot, the badge frame isn't lining up horizontally like tabbarview or others. that looks like a bug.

Reply:
I am very sorry but I didn't understand where is the problem? How the badge frame isn't lining up horizontally?

Reply:
<img width="149" alt="Screen Shot 2021-07-12 at 9 22 20 AM" src="https://user-images.githubusercontent.com/20715435/125322526-bc0f1480-e2f2-11eb-990f-77a314ff79ef.png">

Reply:
sorry i meant vertically

Reply:
@harrieshin Ohh!! you mean vertically but that is actually intentional because one button contains just the image and one button contains just the text and our aim is always to put the badge view touching the top-right corner of the text/image in the button so I believe that this is correct but if you suggest then we can talk to designers from Fluent UI team. Button height changes with different kinds of buttons. Please check the screenshot.
<img width="335" alt="Screenshot 2021-07-13 at 11 37 07 AM" src="https://user-images.githubusercontent.com/33866787/125399940-df0ef680-e3ce-11eb-9471-f110f8573c7e.png">

If the button is of the same type then it will show perfectly.
<img width="338" alt="Screenshot 2021-07-13 at 11 34 56 AM" src="https://user-images.githubusercontent.com/33866787/125399986-f0580300-e3ce-11eb-9c00-c650c7d278c2.png">